### PR TITLE
Default to gcr.io image registry

### PR DESCRIPTION
**Pull request by akutz**
**What this PR does / why we need it**:
This PR changes the behavior of building and pushing images to use the gcr.io image registry.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: NA

**Special notes for your reviewer**:
This PR includes a number of small fixes discovered while implementing the primary feature. However, the changes **are** isolated into individual commits.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: hack/make.sh [hunk 1]
```diff
@@ -14,10 +14,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# posix complaint
```
Comment (frapposelli) [inline hack/make.sh]: s/complaint/compliant

File: hack/make.sh [hunk 2]
```diff
@@ -14,10 +14,21 @@
...
+# When in an interactive terminal add the -t flag so Docker inherits
+# a pseudo TTY. Otherwords SIGINT does not work to kill the container
+# when running this script interactively.
+TERM_FLAGS="-i"
+echo "${-}" | grep -q i && TERM_FLAGS="${TERM_FLAGS}t"
+
+docker run ${TERM_FLAGS} \
+  -v "$(pwd)":/go/src/k8s.io/cloud-provider-vsphere:z \
+  -w /go/src/k8s.io/cloud-provider-vsphere \
+  golang:1.11 make "${@}"
```
Comment (frapposelli) [inline hack/make.sh]: can we parametrize the go version somewhere? it's going to be easier to source it from a known location across all the scripts in the repo.

File: hack/make.sh (hunk 1 above)
Comment (akutz) [reply hack/make.sh]: Fixed

File: hack/make.sh (hunk 2 above)
Comment (akutz) [reply hack/make.sh]: Hi @frapposelli,

Is it used elsewhere? I kind of feel the version of Go should be fixed. We don't necessarily want it to be easy for people to produce binaries with different Go versions. I think it's good that it's fairly assured that the image is the same each time.

**Comment by akutz** [comment]
Hi @frapposelli,

Any other changes I should make?

File: hack/make.sh [hunk 3]
```diff
@@ -14,10 +14,21 @@
...
 # Simple script to offer the ability to run the make directives in a Docker Container.
 # In other words you don't need a Go env on your system to run Make (build, test etc)
 # This script will just bind-mount the source directory into a container under the correct
 # GOPATH and handle all of the Go ENV stuff for you.  All you need is Docker
-docker run -i -v "$PWD":/go/src/k8s.io/cloud-provider-vsphere:z \
-	-w /go/src/k8s.io/cloud-provider-vsphere \
-	golang:1.10 make "$@"
+
+# When in an interactive terminal add the -t flag so Docker inherits
+# a pseudo TTY. Otherwords SIGINT does not work to kill the container
```
Comment (figo) [inline hack/make.sh]: Otherwords -> Otherwise?

**Pull request merged**